### PR TITLE
Implemented International Bank Account flow

### PR DESCRIPTION
**Pull request by shubham1206agra**
### Fixed Issues

$ https://github.com/Expensify/App/issues/51082

### Tests

https://docs.google.com/document/d/19ltca0iKwB00pReokjBf9rEHoi1u8y-2_x-82jMibho/edit?pli=1&tab=t.0#bookmark=id.tf0af0lugbin

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as Tests

### QA Steps

Same as Tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/703a90e7-eb03-44e4-888f-99c5579af135

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/f206f686-a5a9-4293-816d-418ed046d066

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/0578658a-f0a9-415a-a474-8aae5357934f

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/bbf46746-b81c-4be7-ac6b-b0bb35327a6a

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/b4203870-8269-48ad-b2e2-343cb288f80e

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/883547a4-69ea-4723-bbeb-33b2b8701a48

</details>

### CODE REVIEW

**Comment by shawnborton** [comment]
For this screen here:
CleanShot 2024-11-25 at 08 48 27@2x

Is the last input disabled or something? If it's a read-only input, we shouldn't use the input style but rather we should use a style that looks like a push row with no arrow, like so:
CleanShot 2024-11-25 at 08 49 12@2x

**Comment by dannymcclain** [comment]
I agree with Shawn, let's just make it look like a read-only bit of info (like a read-only push input row as Shawn pointed out).

I will note that I think technically that is our disabled input* style though 😕 
*proper input, _not_ push-input

File: src/languages/es.ts [hunk 1]
```diff
@@ -1979,6 +1979,14 @@ const translations = {
             ownershipPercentage: 'Por favor, ingrese un número de porcentaje válido.',
         },
     },
+    addPersonalBankAccount: {
+        countrySelectionStepHeader: "Where's your bank account located?",
```
Comment (DylanDylann) [inline src/languages/es.ts:1983]: @shubham1206agra Please update Spanish translation

File: src/pages/settings/Wallet/InternationalDepositAccount/substeps/AccountType.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    const onAccountTypeSelected = useCallback(() => {
+        if (isEditing && formValues[CONST.CORPAY_FIELDS.ACCOUNT_TYPE_KEY] === currentAccountType) {
+            onNext();
+            return;
+        }
+        if (fieldData.isRequired && !currentAccountType) {
+            return;
+        }
+        FormActions.setDraftValues(ONYXKEYS.FORMS.INTERNATIONAL_BANK_ACCOUNT_FORM, {[CONST.CORPAY_FIELDS.ACCOUNT_TYPE_KEY]: currentAccountType});
```
Comment (DylanDylann) [inline src/pages/settings/Wallet/InternationalDepositAccount/substeps/AccountType.tsx:29]: Why don't we use handleSubmit like other pages?

File: src/pages/settings/Wallet/InternationalDepositAccount/substeps/BankAccountDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                    ErrorUtils.addErrorMessage(errors, fieldName, translate('common.error.fieldRequired'));
+                }
+                fields[fieldName].validationRules.forEach((rule) => {
+                    const regExpCheck = new RegExp(rule.regEx);
+                    if (!regExpCheck.test(values[fieldName])) {
+                        ErrorUtils.addErrorMessage(errors, fieldName, rule.errorMessage);
+                    }
+                });
+            }
+            return errors;
```
Comment (DylanDylann) [inline src/pages/settings/Wallet/InternationalDepositAccount/substeps/BankAccountDetails.tsx:64]: This code is duplicated in many places. Let's DRY it

File: src/pages/settings/Wallet/InternationalDepositAccount/substeps/Confirmation.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    const [corpayFields] = useOnyx(ONYXKEYS.CORPAY_FIELDS);
+
+    const getDataAndGoToNextStep = (values: FormOnyxValues<typeof ONYXKEYS.FORMS.INTERNATIONAL_BANK_ACCOUNT_FORM>) => {
+        setIsSubmitting(true);
+        BankAccounts.createCorpayBankAccount(
+            {...formValues, ...values},
+            corpayFields?.classification ?? '',
+            corpayFields?.destinationCountry ?? '',
+            corpayFields?.preferredMethod ?? '',
+        ).then((response) => {
```
Comment (DylanDylann) [inline src/pages/settings/Wallet/InternationalDepositAccount/substeps/Confirmation.tsx:43]: Should we handle failure case? Currently, there are no feedback if the API failed

Comment (DylanDylann) [reply src/pages/settings/Wallet/InternationalDepositAccount/substeps/Confirmation.tsx:43]: I think we can display a error message in this case

Comment (DylanDylann) [reply src/pages/settings/Wallet/InternationalDepositAccount/substeps/Confirmation.tsx:43]: @shubham1206agra I mean we can handle errors in failureData and maybe display that error on the confirmation page

Comment (DylanDylann) [reply src/pages/settings/Wallet/InternationalDepositAccount/substeps/Confirmation.tsx:43]: There are no need any new component

File: src/hooks/useSubStep/index.ts [hunk 5]
```diff
@@ -1,63 +1,88 @@
...
+export default function useSubStep<TProps extends SubStepProps>({bodyContent, onFinished, startFrom = 0, skipSteps = [], onNextSubStep = () => {}}: UseSubStep<TProps>) {
     const [screenIndex, setScreenIndex] = useState(startFrom);
     const isEditing = useRef(false);
 
+    if (bodyContent.length === skipSteps.length) {
+        throw new Error('All steps are skipped');
+    }
+
+    const lastScreenIndex = useMemo(() => calculateLastIndex(bodyContent.length, skipSteps), [bodyContent.length, skipSteps]);
+
```
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/hooks/useSubStep/index.ts:31]: Hmm, could you help to share problems which you get? I think this way will be cleaner and simpler

File: src/pages/settings/Wallet/InternationalDepositAccount/substeps/Confirmation.tsx (hunk 4 above)
Comment (DylanDylann) [reply src/pages/settings/Wallet/InternationalDepositAccount/substeps/Confirmation.tsx:43]: In createCorpayBankAccount function, let's add corpayFields.errors in failureData like this

https://github.com/Expensify/App/blob/3cc88f5a83a979c5e588e01065083a3d714e69da/src/libs/actions/Card.ts#L101

And then in Confirmation, let's add the error message to Form like here

https://github.com/Expensify/App/blob/3cc88f5a83a979c5e588e01065083a3d714e69da/src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx#L129

File: src/hooks/useSubStep/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/hooks/useSubStep/index.ts:31]: @shubham1206agra 


> I think we should filter out skipSteps from bodyContent and

If we filter out skipSteps from bodyContent, I think we don't need to change anything to logic to navigate to the next step and the previous step.

Could you please detail this point?

>  it will be hard to determine the next or previous step. And the indexes will be jumbled if I want to do a onMove step.

**Comment by DylanDylann** [comment]
@shubham1206agra Bump

**Comment by MonilBhavsar** [comment]
Thanks Shubham!
@DylanDylann over to you

**Comment by shawnborton** [comment]
Much better! Let's get this one over the finish line.

**Comment by DylanDylann** [comment]
@shubham1206agra Could you please check this [comment](https://github.com/Expensify/App/pull/52322/files#r1864798295)?

**Comment by DylanDylann** [comment]
@shubham1206agra I don't have access to manual test docs, could you please give details on the test section on the PR description?

**Comment by DylanDylann** [comment]
@shubham1206agra When changing the country, we need to collect the account number and bank information as we did when changing currency

https://github.com/user-attachments/assets/da93ad28-bb85-45ab-9078-1e5d17766fcf

**Comment by DylanDylann** [comment]
@shubham1206agra Please add save button on currency selector screen

Screenshot 2024-12-09 at 15 37 51

File: src/pages/settings/Wallet/InternationalDepositAccount/utils.ts [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        bankCurrency: internationalBankAccountDraft?.bankCurrency ?? corpayFields?.bankCurrency,
+        accountHolderName: !isEmptyObject(personalDetailsFieldMap?.accountHolderName) ? internationalBankAccountDraft?.accountHolderName ?? fullName : undefined,
+        accountHolderAddress1: !isEmptyObject(personalDetailsFieldMap?.accountHolderAddress1) ? internationalBankAccountDraft?.accountHolderAddress1 ?? street1 : undefined,
+        accountHolderAddress2: !isEmptyObject(personalDetailsFieldMap?.accountHolderAddress2) ? internationalBankAccountDraft?.accountHolderAddress2 ?? street2 : undefined,
+        accountHolderCity: !isEmptyObject(personalDetailsFieldMap?.accountHolderCity) ? internationalBankAccountDraft?.accountHolderCity ?? address?.city : undefined,
+        accountHolderCountry: !isEmptyObject(personalDetailsFieldMap?.accountHolderCountry)
+            ? internationalBankAccountDraft?.accountHolderCountry ?? corpayFields?.bankCountry ?? address?.country ?? latestBankAccount?.bankCountry ?? country ?? ''
```
Comment (DylanDylann) [inline src/pages/settings/Wallet/InternationalDepositAccount/utils.ts:65]: @shubham1206agra This is accountHolderCountry.  Why do we use `corpayFields?.bankCountry`?

**Comment by DylanDylann** [comment]
@shubham1206agra From this [comment](https://github.com/Expensify/App/pull/52322#issuecomment-2498083829), I think we need to fix it on the BE side instead of disable it (or maybe note to do it later)

cc @MonilBhavsar

File: src/pages/settings/Wallet/InternationalDepositAccount/utils.ts (hunk 6 above)
Comment (DylanDylann) [reply src/pages/settings/Wallet/InternationalDepositAccount/utils.ts:65]: From design doc, this field should use the data from private_personalDetails

Screenshot 2024-12-09 at 16 34 34

**Comment by DylanDylann** [comment]
From design doc, we apply offline pattern C for Country/Currency and Confirm Button. For other fields, I think we still need to apply offline pattern B  

https://github.com/user-attachments/assets/30370b1d-412f-4885-93a2-157432f60d79

**Comment by DylanDylann** [comment]
@shubham1206agra 

> @DylanDylann What do you mean here?

Ref: https://github.com/Expensify/App/blob/main/contributingGuides/OFFLINE_UX.md

**Comment by DylanDylann** [comment]
@shubham1206agra What do you think about my other comments?

**Comment by DylanDylann** [comment]
@shubham1206agra 

> @DylanDylann I know the patterns, but can you elaborate where do I need to apply Pattern B?

For all other fields (except country and currency), when users edit these fields offline we will grey out the edited value in confirmation page and its page

But I think we need to confirm with @youssef-lr first because on the design docs, it only says:

> Otherwise, you can complete the form offline up until the [confirmation page](https://docs.google.com/document/d/19ltca0iKwB00pReokjBf9rEHoi1u8y-2_x-82jMibho/edit?pli=1#bookmark=id.j16tjnvfskxq), where Country, Currency, and Confirm are all disabled.

and doesn't mention that we should apply offline pattern B

**Comment by DylanDylann** [comment]
@shubham1206agra Is there any problem here?

**Comment by joekaufmanexpensify** [comment]
@shubham1206agra @DylanDylann I just shared [the manual testing script](https://docs.google.com/document/d/1H1hMnTcMrR8MT6BXz3fCE99eVIurgrHDtaxIH5aW8Ug/edit?tab=t.0#heading=h.nhtsmauf72jq) with you both!

File: src/hooks/useSubStep/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/hooks/useSubStep/index.ts:31]: @shubham1206agra Could you bring your thoughts to this thread?

**Comment by DylanDylann** [comment]
@shubham1206agra Could you give an update on these comments?
https://github.com/Expensify/App/pull/52322#issuecomment-2527271109
https://github.com/Expensify/App/pull/52322#issuecomment-2527190489

**Comment by DylanDylann** [comment]
@shubham1206agra @Expensify/design  When clicking the confirm button, should we display a loading icon instead of "Confirm" while waiting response from BE? I think it will be better from UX perspective




https://github.com/user-attachments/assets/525e0e1a-c726-446e-a752-36b16a29ffe9

**Comment by DylanDylann** [comment]
I saw it before

**Comment by shawnborton** [comment]
> When clicking the confirm button, should we display a loading icon instead of "Confirm" while waiting response from BE? I think it will be better from UX perspective

That makes total sense to me and feels consistent with how we handle these sorts of things elsewhere. Let's update that.

**Comment by dannymcclain** [comment]
Agree with Shawn about the loader/spinner.

**Comment by DylanDylann** [comment]
Waiting for @shubham1206agra to update above comments:
- https://github.com/Expensify/App/pull/52322#issuecomment-2547932716
- https://github.com/Expensify/App/pull/52322#issuecomment-2548006697

**Comment by DylanDylann** [comment]
@shubham1206agra waiting for you : https://github.com/Expensify/App/pull/52322#issuecomment-2550232682

**Comment by DylanDylann** [comment]
Ops, It seems I mentioned a wrong link

**Comment by DylanDylann** [comment]
Please check these comments: 
1. https://github.com/Expensify/App/pull/52322/files#r1875657445
2. https://github.com/Expensify/App/pull/52322#issuecomment-2548006697
3. https://github.com/Expensify/App/pull/52322#issuecomment-2527271109

**Comment by DylanDylann** [comment]
Ops, we got a failed lint but It caused by our new rule caused it. I think we can ignore it in this PR because this PR already be so big. Wdyt?

cc @MonilBhavsar @shubham1206agra

**Comment by DylanDylann** [comment]
@shubham1206agra I think we need to confirm with the design team if we want to change the original design. But we can do it in a follow up

> @DylanDylann Added header content, but unable to add button as this requires major changes to CurrencySelectionList.

**Comment by DylanDylann** [comment]
> @DylanDylann The lint issue will spread to other files as the changes to function will cascade to different files. Its best to leave it for now.

For this one, I will leave the final decision to @MonilBhavsar

**Review by DylanDylann — APPROVED** [review]
The rest looks good to me 💯

**Review by DylanDylann — APPROVED** [review]

**Comment by MonilBhavsar** [comment]
@shubham1206agra the ESLint tests are failing

File: src/hooks/useSubStep/index.ts [hunk 7]
```diff
@@ -1,63 +1,88 @@
...
     const prevScreen = useCallback(() => {
-        const prevScreenIndex = screenIndex - 1;
+        let decrementNuber = 1;
```
Comment (MonilBhavsar) [inline src/hooks/useSubStep/index.ts:33]:
```suggestion
        let decrementNumber = 1;
```

File: src/hooks/useSubStep/index.ts [hunk 8]
```diff
@@ -1,63 +1,88 @@
...
+        while (screenIndex - decrementNuber >= 0 && skipSteps.includes(screenIndex - decrementNuber)) {
```
Comment (MonilBhavsar) [inline src/hooks/useSubStep/index.ts:34]:
```suggestion
        while (screenIndex - decrementNumber >= 0 && skipSteps.includes(screenIndex - decrementNuber)) {
```

File: src/hooks/useSubStep/index.ts [hunk 9]
```diff
@@ -1,63 +1,88 @@
...
+            decrementNuber += 1;
```
Comment (MonilBhavsar) [inline src/hooks/useSubStep/index.ts:35]:
```suggestion
            decrementNumber += 1;
```

File: src/hooks/useSubStep/index.ts [hunk 10]
```diff
@@ -1,63 +1,88 @@
...
+        }
+        const prevScreenIndex = screenIndex - decrementNuber;
```
Comment (MonilBhavsar) [inline src/hooks/useSubStep/index.ts:37]:
```suggestion
        const prevScreenIndex = screenIndex - decrementNumber;
```

File: src/hooks/useSubStep/index.ts (hunk 8 above)
Comment (MonilBhavsar) [inline src/hooks/useSubStep/index.ts:34]: Can you please add a comment here

File: src/libs/actions/BankAccounts.ts [hunk 11]
```diff
@@ -62,6 +62,12 @@ function clearPlaid(): Promise<void | void[]> {
     return Onyx.set(ONYXKEYS.PLAID_DATA, CONST.PLAID.DEFAULT_DATA);
 }
 
+function clearInternationalBankAccount() {
+    return clearPlaid()
```
Comment (MonilBhavsar) [inline src/libs/actions/BankAccounts.ts:66]: Why we clear Plaid? I don't think it's required, as we don't interact with Plaid in this workflow

File: src/pages/settings/Wallet/InternationalDepositAccount/substeps/AccountHolderInformation.tsx [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+    if ((field.valueSet ?? []).length > 0) {
+        return ValuePicker;
+    }
+    if (CONST.CORPAY_FIELDS.SPECIAL_LIST_REGION_KEYS.includes(field.id)) {
+        return ValuePicker;
+    }
+    if (CONST.CORPAY_FIELDS.SPECIAL_LIST_ADDRESS_KEYS.includes(field.id)) {
+        return AddressSearch;
+    }
+    if (field.id === 'accountHolderCountry') {
```
Comment (MonilBhavsar) [inline src/pages/settings/Wallet/InternationalDepositAccount/substeps/AccountHolderInformation.tsx:31]: We could make this a constant

File: src/types/onyx/CorpayFields.ts [hunk 13]
```diff
@@ -0,0 +1,135 @@
+/**
+ *
+ */
+type CorpayFormField = {
+    /**
+     *
```
Comment (MonilBhavsar) [inline src/types/onyx/CorpayFields.ts:6]: Missing values?

File: tests/unit/useSubStepTest.tsx [hunk 14]
```diff
@@ -93,3 +100,191 @@ describe('useSubStep hook', () => {
         expect(newComponentToRender).toBe(MockSubStepComponent2);
     });
 });
+
+describe('useSubStep hook with skipSteps', () => {
```
Comment (MonilBhavsar) [inline tests/unit/useSubStepTest.tsx:104]: Thanks for adding tests!
Can we please document these tests with Given... When... Then... format https://github.com/Expensify/App/tree/main/tests#documenting-tests

**Comment by MonilBhavsar** [comment]
Bug: After clicking on confirm, bank account was added successfully, but modal didn't close


Screenshot 2024-12-30 at 1 51 06 PM

File: src/libs/API/parameters/VerifyIdentityForBankAccountParams.ts [hunk 15]
```diff
@@ -1,6 +1,6 @@
 type VerifyIdentityForBankAccountParams = {
     bankAccountID: number;
     onfidoData: string;
-    policyID?: string;
+    policyID: string;
```
Comment (MonilBhavsar) [inline src/libs/API/parameters/VerifyIdentityForBankAccountParams.ts:4]: NAB, We could probably revert this change as the code works with empty policyID

**Review by MonilBhavsar — APPROVED** [review]

**Comment by kavimuru** [comment]
@shubham1206agra @MonilBhavsar What our QA team should be validating here? Could you help with this?

**Comment by MonilBhavsar** [comment]
@kavimuru the QA steps are in this doc https://docs.google.com/document/d/1H1hMnTcMrR8MT6BXz3fCE99eVIurgrHDtaxIH5aW8Ug/edit?usp=sharing

**Comment by MonilBhavsar** [comment]
Please tag me and @shubham1206agra if any regression is coming out from this PR. We'll handle that in a followup PR. Thanks!

**Comment by Beamanator** [comment]
reverted - see https://github.com/Expensify/App/pull/54729
